query-engine/query-engine-node-api/src/engine.rs

```diff
@@ -242,35 +241,34 @@ impl QueryEngine {
             };
 
             let engine = async move {
-                let executor_fut = tokio::spawn(async move {
-                    // We only support one data source & generator at the moment, so take the first one (default not exposed yet).
-                    let data_source = arced_schema
-                        .configuration
-                        .datasources
-                        .first()
-                        .ok_or_else(|| ApiError::configuration("No valid data source found"))?;
-
-                    let preview_features = arced_schema.configuration.preview_features();
-
-                    let executor = load_executor(data_source, preview_features, &url).await?;
-                    let connector = executor.primary_connector();
-                    connector.get_connection().await?;
-                    crate::Result::<_>::Ok(executor)
-                });
+                // We only support one data source & generator at the moment, so take the first one (default not exposed yet).
+                let data_source = arced_schema
+                    .configuration
+                    .datasources
+                    .first()
+                    .ok_or_else(|| ApiError::configuration("No valid data source found"))?;
+
+                let preview_features = arced_schema.configuration.preview_features();
+
+                let executor = load_executor(data_source, preview_features, &url).await?;
```

Comment:
This is no longer parallelized. It doesn't really matter IMO since it doesn't perform anything expensive. I actually see no reason why this should be async.